packages/@aws-cdk/aws-cloudtrail/lib/cloudtrail.ts

```diff
@@ -235,7 +236,7 @@ export class Trail extends Resource {
 
     let logsRole: iam.IRole | undefined;
 
-    if (props.sendToCloudWatchLogs) {
+    if (props.sendToCloudWatchLogs || props.cloudWatchLogGroup || props.cloudWatchLogsRetention) {
```

Comment:
We should throw an error if `sendToCloudWatchLogs` is false and either `cloudWatchLogGroup` or `cloudWatchLogsRetention` is provided. In this case, they could explicitly set it as false and it would still end up creating logs because they added another one of these props.